src/main/java/org/jitsi/videobridge/Conference.java

```diff
@@ -435,9 +436,50 @@ void dominantSpeakerChanged()
             broadcastMessage(
                     createDominantSpeakerEndpointChangeEvent(
                         dominantSpeaker.getID()));
+            if (getEndpointCount() >= 3)
+            {
+                double senderRtt = getRtt(dominantSpeaker);
+                double maxReceiverRtt = getMaxReceiverRtt(dominantSpeaker.getID());
+                // We add an additional 10ms delay to reduce the risk of the keyframe arriving
+                // too early
+                double keyframeDelay = maxReceiverRtt + senderRtt + 10;
+                if (logger.isDebugEnabled())
+                {
+                    logger.debug("Scheduling keyframe request from " + dominantSpeaker.getID() + " after a delay" +
+                            "of " + keyframeDelay + "ms");
+                }
+                TaskPools.SCHEDULED_POOL.schedule((Runnable) dominantSpeaker::requestKeyframe, (long)keyframeDelay, TimeUnit.MILLISECONDS);
+            }
+        }
+    }
+
+    private double getRtt(AbstractEndpoint endpoint)
+    {
+        if (endpoint instanceof Endpoint)
+        {
+            Endpoint localDominantSpeaker = (Endpoint)endpoint;
+            return localDominantSpeaker.getTransceiver().getTransceiverStats().getEndpointConnectionStats().getRtt();
```

Comment:
this (and below) call to get the endpoint's RTT is a bit gnarly.  i considered adding a `getRtt` function to `Endpoint`, but it felt a little weird to bubble up that one stat.  now that i'm seeing it again after a day i think it's probably better to add the `getRtt` function to `Endpoint`, so i'll do that.